processing/data_collection/gazette/spiders/rj_duque_de_caxias.py

```diff
@@ -0,0 +1,121 @@
+import datetime as dt
+import re
+import scrapy
+
+from gazette.items import Gazette
+from dateparser import parse
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class RjDuqueDeCaxiasSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3301702"
+    name = "rj_duque_de_caxias"
+    allowed_domains = ["duquedecaxias.rj.gov.br"]
+    start_urls = ["http://duquedecaxias.rj.gov.br/portal/boletim-oficial.html"]
+
+    def parse(self, response):
+        current_year = dt.date.today().year
+
+        pages = response.xpath("//center//div[1]//div//a/@href").extract()
+        page_urls = [response.urljoin(url) for url in pages]
+
+        years = response.xpath("//center//div[1]//div/text()").re("(\d{4})")
+
+        self.parse_html_indexed_page(response, True, current_year)
+
+        for year, url in zip(years, page_urls):
+            request = scrapy.Request(url=url, callback=self.parse_page)
+            request.meta["year"] = year
+            yield request
+
+    def parse_page(self, response):
+        year = response.meta["year"]
+        if int(year) > 2016:
+            self.parse_html_indexed_page(response, False, year)
+        else:
+            months = [
+                "01-Janeiro",
+                "02-Fevereiro",
+                "03-Marco",
+                "04-Abril",
+                "05-Maio",
+                "06-Junho",
+                "07-Julho",
+                "08-Agosto",
+                "09-Setembro",
+                "10-Outubro",
+                "11-Novembro",
+                "12-Dezembro",
+            ]
+
+            for month in months:
+                url = response.urljoin(month)
+                request = scrapy.Request(url=url, callback=self.parse_directory_page)
+                request.meta["year"] = year
+                request.meta["month"] = month
+                yield request
+
+    def parse_directory_page(self, response):
+        month = response.meta["month"]
+        year = response.meta["year"]
+        pdf_names = response.xpath("//a/attribute::href").re("(.*).pdf")
+
+        urls = ["%s%s" % (response.urljoin(url), ".pdf") for url in pdf_names]
+
+        dates = []
+        for pdf in pdf_names:
+            dat = "%s/%s/%s" % (pdf[-2:], month[:2], year)
```

Comment:
Use f string instead of .format or %.  (The why is [here](http://www.anapaulagomes.me/pt-br/python-drops-f-strings/) - TLDR: More readable and faster)